### PR TITLE
[examples] add an example project to show test strategies

### PR DESCRIPTION
**Pull request by sebsto**
This new example project show four testing strategies for Swift Lambda function 

- Unit testing the business logic (not specific to Swift Lambda) 
- Integration testing the handler method 
- Local invocation with the Swift Lambda Runtime 
- Local invocation with SAM 

**[IMPORTANT]**
To allow testing the handler, I had to change visibility of a method in the Runtime project.  This method is clearly marked for testing only, so it should not be a problem. Happy to read feedback and discuss however.

### CODE REVIEW

File: Sources/AWSLambdaRuntimeCore/LambdaContext.swift [hunk 1]
```diff
@@ -120,8 +120,10 @@ public struct LambdaContext: CustomDebugStringConvertible, Sendable {
         "\(Self.self)(requestID: \(self.requestID), traceID: \(self.traceID), invokedFunctionARN: \(self.invokedFunctionARN), cognitoIdentity: \(self.cognitoIdentity ?? "nil"), clientContext: \(self.clientContext ?? "nil"), deadline: \(self.deadline))"
     }
 
-    /// This interface is not part of the public API and must not be used by adopters. This API is not part of semver versioning.
-    package static func __forTestsOnly(
+    /// This interface is not part of the public API and must not be used by adopters.
+    /// It is declared public to allow users to unit test their handler.
+    /// This API is not part of semver versioning.
+    public static func __forTestsOnly(
```
Comment (0xTim) [inline Sources/AWSLambdaRuntimeCore/LambdaContext.swift:126]: Why are we making this public if it can be visible with `@testable`?

Comment (0xTim) [reply Sources/AWSLambdaRuntimeCore/LambdaContext.swift:126]: If we want it public, we might want to consider introducing a `LambdaTesting` target to keep it separate

**Review by 0xTim — COMMENTED** [review]
Some queries around testing infrastructure

File: Sources/AWSLambdaRuntimeCore/LambdaContext.swift (hunk 1 above)
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/LambdaContext.swift:126]: @0xTim 

You're correct, I was importing `@testable import AWSLambdaRuntime` but I needed `AWSLambdaRuntimeCore`
Thank you !

**Comment by sebsto** [comment]
Note that the integration tests fail pending merging of https://github.com/swift-server/swift-aws-lambda-runtime/pull/441

**Review by 0xTim — APPROVED** [review]

**Pull request merged**